src/realm/sync/noinst/client_reset_recovery.cpp

```diff
@@ -706,6 +707,24 @@ void RecoverLocalChangesetsHandler::operator()(const Instruction::AddTable& inst
     // Rely on InstructionApplier to validate existing tables
     StringData class_name = get_string(instr.table);
     try {
+        auto table = table_for_class_name(class_name);
+        InstructionApplier::operator()(instr);
+
+        // if the table already existed then no instruction was
```

Comment:
What would be a scenario where this can happen?

Reply:
It can happen in this scenario:

1. A client opens and sync a Realm with table `A`
2. The Realm is closed and the server is terminated so that the next time this client connects, it will be reset.
3. After some time, the user installs a new version of the app which now has both table `A` and `B` in the schema.
4. Open the app and attempt to sync. This creates a transaction "offline" that includes the new AddTable instruction for `B`.
5. Client reset with recovery is initiated.
6. The AddTable instruction for table `B` is recovered here, and that table already exists on disk.
    - Before the changes in this PR, the AddTable instruction would not be sent to the server, which means that any updates to that table would be rejected with a fatal error unless the table had been manually added to the server side schema.
    - But with these changes, the server may accept the AddTable instruction (if in dev mode) and the recovery will be successful.

Reply:
At step 6, why wouldn't the AddTable instruction be sent to the server? Every instruction recovered should be replicated and uploaded to the server right?

Reply:
During a client reset with recovery, the history cannot be sent verbatim to the server because some instructions are not recoverable, and others (such as operations on lists) must be modified. To do that, we pass the history through the  `RecoverLocalChangesetsHandler` which is a custom `InstructionApplier` so that the instructions are replayed and the new history that is produced from that is what is sent to the server. In that case, the table exists but replaying the "AddTable" instruction wouldn't add the history part that we want. This code is covered by the "Recover: additive schema changes are recovered in dev mode" test.

Reply:
got it. I see schema instructions are special in that regard.